pkg/controller/version/docker_test.go

```diff
@@ -0,0 +1,53 @@
+package version
+
+import (
+	"testing"
+
+	"github.com/Dynatrace/dynatrace-activegate-operator/pkg/controller/parser"
+	"github.com/containers/image/v5/transports/alltransports"
+	"github.com/containers/image/v5/types"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMakeSystemContext(t *testing.T) {
+	versionChecker := NewDockerVersionChecker(
+		"localhost.com/image:1234",
+		"localhost.com/image:1234",
+		nil)
+
+	assert.NotNil(t, versionChecker)
+
+	reference, err := alltransports.ParseImageName("docker://localhost/image:1234")
+	assert.NoError(t, err)
+	assert.NotNil(t, reference)
+
+	noAuth := versionChecker.makeSystemContext(reference.DockerReference())
+	assert.Equal(t, types.SystemContext{}, *noAuth)
+
+	type auth struct {
+		Username string
+		Password string
+	}
+	auths := make(map[string]struct {
+		Username string
+		Password string
+	})
```

Comment:
```suggestion
	auths := make(map[string]auth)
```

Reply:
This results in the following error message when compiling:
`cannot use auths (type map[string]auth) as type map[string]struct { Username string; Password string } in field value`

Reply:
I created a new type `DockerConfigAuth` to simplify those calls